exercises/list-ops/src/test/java/ListOpsTest.java

```diff
@@ -1,316 +1,186 @@
-        final List<Integer> actual = ListOps.concat(EMPTY_LIST, EMPTY_LIST);
-
-        assertNotNull(actual);
-        assertTrue(actual.isEmpty());
+    @Test
+    public void testTransformingNonEmptyList() {
+        assertEquals(
+                Arrays.asList(2, 4, 6, 8),
+                ListOps.map(Arrays.asList(1, 3, 5, 7), integer -> integer + 1 ));
     }
 
-    @Test
     @Ignore("Remove to run test")
-    public void shouldConcatenateOneEmptyAndOneNonEmptyLists() {
-        final List<Integer> list
-                = Collections.unmodifiableList(
-                        Arrays.asList(0, 1, 2, 3, 4)
-                );
-        final List<Integer> actual = ListOps.concat(list, EMPTY_LIST);
-        final List<Integer> expected
-                = Arrays.asList(0, 1, 2, 3, 4);
-
-        assertNotNull(actual);
-        assertFalse(actual.isEmpty());
-        assertEquals(expected, actual);
+    @Test
+    public void testFoldLeftOnEmptyList() {
+        assertEquals(
+                new Double(2.0), // Boxing required for method overload disambiguation.
+                ListOps.foldLeft(
+                        Collections.<Double>emptyList(),
+                        2.0,
+                        (accumulation, current) -> current * accumulation));
     }
 
-    @Test
     @Ignore("Remove to run test")
-    public void shouldConcatenateOneNonEmptyAndOneEmptyLists() {
-        final List<Integer> list
-                = Collections.unmodifiableList(
-                        Arrays.asList(0, 1, 2, 3, 4)
-                );
-        final List<Integer> actual = ListOps.concat(EMPTY_LIST, list);
-        final List<Integer> expected
-                = Arrays.asList(0, 1, 2, 3, 4);
-
-        assertNotNull(actual);
-        assertFalse(actual.isEmpty());
-        assertEquals(expected, actual);
+    @Test
+    public void testFoldLeftWithDirectionIndependentOperationOnNonEmptyList() {
+        assertEquals(
+                new Integer(15), // Boxing required for method overload disambiguation.
+                ListOps.foldLeft(
+                        Arrays.asList(1, 2, 3, 4),
+                        5,
+                        (accumulation, current) -> current + accumulation));
     }
 
-    @Test
     @Ignore("Remove to run test")
-    public void shouldConcatenateTwoListsWithSameElements() {
-        final List<Integer> list1 = Collections.unmodifiableList(
-                Arrays.asList(0, 1, 2, 3, 4)
-        );
-        final List<Integer> list2 = Collections.unmodifiableList(
-                Arrays.asList(1, 2, 3, 4, 5, 6)
-        );
-        final List<Integer> expected
-                = Arrays.asList(0, 1, 2, 3, 4, 1, 2, 3, 4, 5, 6);
-        final List<Integer> actual = ListOps.concat(list1, list2);
-
-        assertNotNull(actual);
-        assertFalse(actual.isEmpty());
-        assertEquals(expected, actual);
+    @Test
+    public void testFoldLeftWithDirectionDependentOperationOnNonEmptyList() {
```

Comment:
Ok, so the difference between these two cases is due to some subtleties of how the fold function parameter is typed. In particular, note that it is of type

```
BiFunction<U, T, U>
```

for the left fold, and of type

```
BiFunction<T, U, U>
```

for the right fold. Since `T` is the type of the list entries, and `U` is the type of the accumulated value, this means that if we pass the anonymous function `(x, y) -> x / y` to both folds, we're actually performing slightly different calculations. See [this upstream comment and the ones following](https://github.com/exercism/problem-specifications/issues/826#issuecomment-307626155) for some relevant chatter. So what we're actually doing is the following:

---

If folding from the left:

Step 0: accumulation = 5
Step 1: accumulation = a / c = 5 / 2 = 2
Step 2: accumulation = a / c = 2 / 5 = 0

If folding from the right:

Step 0: accumulation = 5
Step 1: accumulation = c / a = 5 / 5 = 1
Step 2: accumulation = c / a = 2 / 1 = 2

I updated the function signatures in the test suite to use generic parameter labels `x` and `y` rather than `accumulation` and `current` to make it clearer that we're using the same anonymous function in both cases. It's up to the user to infer from the (provided) type signature how those parameters should be populated. I update the HINTS file to provide a direct link to the section of the fold Wikipedia page relevant to interpretation of these signatures.